### PR TITLE
chore: refactor product images tests and model implementation to be more logical and fix some bugs"

### PR DESCRIPTION
**Pull request by frederickfogerty**
This PR addresses some inconsistencies in the testing and product images implementation

## Before

- Tests: hard to distinguish in testing data between SF data and imgix attribute data
- Tests: was hard to distinguish the context behind tests (whether custom attribute data was set, whether imgix base URL preference was set)
- Tests: not all edge cases were tested (e.g. requesting multiple images when no imgix base URL preference was set caused a `TypeError`)
- Product Images Model: there was a bug where requesting multiple images when no imgix base URL preference was set would cause `TypeError`
- Product Images Model: the organisation made it difficult to reason about the mode of operation, when there were distinctly three modes of operation: using custom attribute data, using SF data w/ imgix base URL preference set, and falling back to SF data when no imgix base URL preference was set
- Product Images Model: `absURL` data returned was not modified, causing potentially inconsistencies and bugs downstream depending on usage of `url` and `absURL`

## After

- Tests: SF data encoded with `sf_` prefix, and imgix attribute data prefixed with `imgix_`. Also the imgix domain for custom attribute data was changed to `customImgixURL`. This allows the tests to better test that the data is being pulled from the right place. E.g. before the implementation could have pulled the imgix domain from the custom attribute, but the path from the SF data, and the test would still would have passed
- Tests: more edge cases were added
- Tests: some stubbed tests were added to be implemented later
- Product Image Model: requesting multiple images when no imgix base URL preference is set no longer causes an error
- Product Image Model: refactored to show distinct differences between three modes of operation
- Product Image Model: `absURL` now points to imgix url, when it exists. **Edit:** confirmed this with Hasan and this is correct behaviour

### CODE REVIEW

**Comment by frederickfogerty** [comment]
Current dependencies on/for this PR:
* next
  * **PR #189** 
    * **PR #188**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: test/unit/int_imgix_products_sfra/models/product/productImages.js [hunk 1]
```diff
@@ -122,161 +121,270 @@ describe("productImages", function () {
...
+      it("medium image");
+      it("large image");
+      it("original size image");
+    });
+
+    describe("should resize image from custom attribute", () => {
+      it("small image");
+      it("medium image");
+      it("large image");
+      it("original size image");
```
Comment (luqven) [inline test/unit/int_imgix_products_sfra/models/product/productImages.js:387]: Assuming this is placeholder?

**Review by luqven — APPROVED** [review]
LGTM nice QOL changes to the model and tests. Much easier to follow now.
